lib/galaxy/datatypes/binary.py

```diff
@@ -978,14 +980,14 @@ def sniff(self, filename):
         return False
 
     def set_meta(self, dataset, overwrite=True, **kwd):
-        super(Anndata, self).set_meta(dataset, overwrite=overwrite, **kwd)
-        try:
+            super(Anndata, self).set_meta(dataset, overwrite=overwrite, **kwd)
             with h5py.File(dataset.file_name, 'r') as anndata_file:
                 dataset.metadata.title = util.unicodify(anndata_file.attrs.get('title'))
                 dataset.metadata.description = util.unicodify(anndata_file.attrs.get('description'))
                 dataset.metadata.url = util.unicodify(anndata_file.attrs.get('url'))
                 dataset.metadata.doi = util.unicodify(anndata_file.attrs.get('doi'))
                 dataset.creation_date = util.unicodify(anndata_file.attrs.get('creation_date'))
+                dataset.metadata.shape = util.unicodify(anndata_file.attrs.get('shape'))
```

Comment:
`unicodify` in essence turns bytestrings to unicode strings, passing through `None`. `anndata_file.attrs.get('shape')` should be `None` or a tuple, of integers, right ? If you get a tuple from `anndata_file.attrs.get('shape')`  and you pass it into unicodify you'll get a string back out:
```python
In [2]: from galaxy.util import unicodify

In [3]: unicodify((2,2))
Out[3]: '(2, 2)'
```
So that makes me think we have no test yet for a file that specifies the shape.
I would guess that the following is closer to what you want to do:

```suggestion
                dataset.metadata.shape = anndata_file.attrs.get('shape') or dataset.metadata.shape
```
That means you can drop line 1054/1055.

If you decide not to wrap this part in try/except you need to dedent the set_meta function body by 4 spaces.

Reply:
I committed the line and unindented the region, but I kept line 1054/55 since I don't quite see how we're handling cases which have no obvious shape assignment but do have observations and variables.